### PR TITLE
Remove all remaining usages of Boundary.

### PR DESCRIPTION
**Pull request by drwells**
Requiescat in pace

                                        March 4, 1998 - May 12, 2018

`Boundary` was added in 1846395, the eighth commit (which also added `Quadrature`, `FEValues`, etc...). I'm a little sad to see one of the original pieces of the library retired.

In a sense, Manifold is not unlike the original design. In the beginning we had the `in_between` function:

```cpp
template <int dim>
class Boundary {
  public:
				     /**
				      *  Typedef an array of the needed number
				      *  of old points.
				      */
    typedef const Point<dim>* PointArray[1<<(dim-1)];

				     /**
				      *  This function calculates the position
				      *  of the new vertex.
				      */
    virtual Point<dim> in_between (const PointArray &neighbors) const = 0;
};

/**
    Specialisation of \Ref{Boundary}<dim>, which places the new point right
    into the middle of the given points. The middle is defined as the
    arithmetic mean of the points.
    */
template <int dim>
class StraightBoundary : public Boundary<dim> {
  public:
				     /**
				      *  This function calculates the position
				      *  of the new vertex.
				      */
    virtual Point<dim> in_between (const PointArray &neighbors) const {
      Point<dim> p;
      for (int i=0; i<(1<<(dim-1)); ++i)
	p += *neighbors[i];
      p /= (1<<(dim-1))*1.0;
      return p;
    };
};
```

which is essentially `Manifold::get_intermediate_point` without the weight.

### CODE REVIEW

**Comment by drwells** [comment]
You're right, of course. There is no way to do higher-order methods without correct geometry descriptions, and this would not be possible without the massive effort by @luca-heltai :)

File: doc/doxygen/headers/boundary.h [hunk 1]
```diff
@@ -1,131 +0,0 @@
...
-// Public License as published by the Free Software Foundation; either
-// version 2.1 of the License, or (at your option) any later version.
-// The full text of the license can be found in the file LICENSE at
-// the top level of the deal.II distribution.
-//
-// ---------------------------------------------------------------------
-
-
-/**
- * @defgroup boundary Boundary and manifold description for triangulations
```
Comment (bangerth) [inline doc/doxygen/headers/boundary.h]: was there anything worth saving from this file?

**Comment by luca-heltai** [comment]
@bangerth , @drwells , without your continuous efforts and support, this big change would have never gotten this far. Big thanks should go to the two of you.

**Comment by bangerth** [comment]
On 05/15/2018 07:42 PM, Luca Heltai wrote:
> @bangerth <https://github.com/bangerth> , @drwells 
> <https://github.com/drwells> , without your continuous efforts and support, 
> this big change would have never gotten this far. Big thanks should go to the 
> two of you.

I didn't do anything. @drwells deserves the credit, and everyone else who 
helped with patches.

(Though I do think that we ought to finish that paper at one point or other!)

File: doc/doxygen/headers/boundary.h (hunk 1 above)
Comment (bangerth) [reply doc/doxygen/headers/boundary.h]: OK. So then this is ready to merge, right? We can handle the glossary entry in a separate patch.

**Comment by luca-heltai** [comment]
I'll pull the trigger here. And it will be an honour.

**Pull request merged**